### DISCUSSION TITLE
v1.96.0

### DISCUSSION THREAD

**Discussion by alextran1502** [immich-app/immich#7478] [Announcements]
# v1.96.0

## Highlights

Welcome to release `v1.96.0` of Immich. This release adds more enhancements for search, optimizes the mobile app and includes a handful of bug fixes. Some of the key highlights include:

- Search filter form improvement
- Search by location when editing the place for the asset.
- Optimization of asset rendering on the mobile app.
- Slideshow options (web)
- Notable fix (mobile app): Stop sending users to the login page for unrelated authentication errors.
- Notable fix: Smart Search when using OpenVINO.

### Improved Search Form

After receiving users' feedback, we make the search filter a little more compact with minor UI/UX changes to make the process of selecting/deselecting the filter's content more intuitive.

![search-filter](https://github.com/immich-app/immich/assets/27055614/5d632481-1cd7-47a2-a576-9182cd6c57b3)

### Search location when editing place for the asset.

We have implemented the lookup-on-type mechanism to quicky search for a location on the map when adding or edding location of the asset. This process happens entirely on your server

https://github.com/immich-app/immich/assets/27055614/7960d897-fc85-44f5-bbb0-03c6d16effa9

### [Web] Slideshow options

We added additional options for the slideshow feature on the web, so you can make it fit your needs when showing of your gallery

![image](https://github.com/immich-app/immich/assets/27055614/ff5f9d61-1a2f-47b9-a07f-8823a8c6185c)

### Optimized asset rendering on the mobile app

We add a blur thumbnail loading step to help the transition from black/white placeholder to the image a more pleasant experience and fix a bug that sometimes the HTTP client would timeout and show the broken icon on the thumbnail.

https://github.com/immich-app/immich/assets/27055614/38c77700-4666-4686-bce2-8229203978d6

And as always, bugs are fixed, and many other improvements also come with this release.

Please consider supporting the project.

## Support

<p align="center">
<img src="https://media.giphy.com/media/LStqgGESXW8XnuCv5y/giphy.gif" width="250" title="SUPPORT THE PROJECT!"> 
</p>

If you find the project helpful, you can support Immich via the following channels.

- Monthly donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502)
- One-time donation via [GitHub Sponsors](https://github.com/sponsors/alextran1502?frequency=one-time&sponsor=alextran1502)
- [Librepay](https://liberapay.com/alex.tran1502/)
- [buymeacoffee](https://www.buymeacoffee.com/altran1502)
- Bitcoin: 3QVAb9dCHutquVejeNXitPqZX26Yg5kxb7

It is a great way to let me know that you want me to continue developing and working on this project for years to come.

### 🗄️ Server
* chore(server): change transcode default to accept all supported audio codecs by @mertalev in https://github.com/immich-app/immich/pull/7283
* fix(server): search with same face multiple times by @michelheusschen in https://github.com/immich-app/immich/pull/7306
* fix(server, web): people page by @martabal in https://github.com/immich-app/immich/pull/7319
* fix(server): fix log for setting other vector extension by @mertalev in https://github.com/immich-app/immich/pull/7325
* perf(server): optimize mapAsset by @michelheusschen in https://github.com/immich-app/immich/pull/7331
* fix(server): out of memory when unstacking assets by @michelheusschen in https://github.com/immich-app/immich/pull/7332
* fix(server): exclude archived assets from orphaned files by @michelheusschen in https://github.com/immich-app/immich/pull/7334
* feat(server, web): search location by @martabal in https://github.com/immich-app/immich/pull/7139
* fix(server): storage template migration not working by @mertalev in https://github.com/immich-app/immich/pull/7414
* fix(server): truncate embedding tables by @mertalev in https://github.com/immich-app/immich/pull/7449
* feat(server): easy RKMPP video encoding by @fyfrey in https://github.com/immich-app/immich/pull/7460
### 📱 Mobile
* refactor(mobile): Use hooks to manage Chewie controller for video by @martyfuhry in https://github.com/immich-app/immich/pull/7008
* fix(deps): update dependency geolocator to v11 by @renovate in https://github.com/immich-app/immich/pull/7249
* fix(deps): update dependency flutter_udid to v3 by @renovate in https://github.com/immich-app/immich/pull/7248
* fix(mobile): don't crop memories in landscape mode by @hrdl-github in https://github.com/immich-app/immich/pull/6907
* refactor(mobile): move error details to separate DB column by @rovo89 in https://github.com/immich-app/immich/pull/6898
* fix(mobile): remove log message counter by @rovo89 in https://github.com/immich-app/immich/pull/6865
* fix(mobile): Stop sending app to login page for unrelated auth errors by @martyfuhry in https://github.com/immich-app/immich/pull/7383
* refactor(mobile): Use ImmichThumbnail and local thumbnail image provider by @martyfuhry in https://github.com/immich-app/immich/pull/7279
* refactor(mobile): Uses blurhash for memory card instead of blurred thumbnail by @martyfuhry in https://github.com/immich-app/immich/pull/7469
### 🖥️ Web
* fix(web): search filter form events by @michelheusschen in https://github.com/immich-app/immich/pull/7285
* fix(web): back button for gallery viewer by @michelheusschen in https://github.com/immich-app/immich/pull/7250
* fix(web): small issues everywhere by @martabal in https://github.com/immich-app/immich/pull/7207
* Localize the output of the library count to make it more readable by @MarcelEeken in https://github.com/immich-app/immich/pull/7305
* feat(web): search improvements and refactor by @michelheusschen in https://github.com/immich-app/immich/pull/7291
* feat(web): rework combobox and add clear button by @michelheusschen in https://github.com/immich-app/immich/pull/7317
* feat(web): search albums by @martabal in https://github.com/immich-app/immich/pull/7322
* feat(web): add an option to change the date formats by @martabal in https://github.com/immich-app/immich/pull/7174
* Enhancement: Revamp onboarding message for Storage Template Engine - Web by @RavidYael in https://github.com/immich-app/immich/pull/7339
* refactor(web): sidebar settings by @jrasm91 in https://github.com/immich-app/immich/pull/7344
* fix: upgrade SvelteKit to 2.5.1 by @benmccann in https://github.com/immich-app/immich/pull/7351
* fix(web): consistent combobox style + improve color contrast by @michelheusschen in https://github.com/immich-app/immich/pull/7353
* feat(web): configure slideshow by @martabal in https://github.com/immich-app/immich/pull/7219
* fix(web): dark mode uploading font color by @sebbi08 in https://github.com/immich-app/immich/pull/7372
* Web: Revamp message for Storage Template Engine in admin pannel by @AndyPro720 in https://github.com/immich-app/immich/pull/7359
* feat(web): improve search filter design by @michelheusschen in https://github.com/immich-app/immich/pull/7367
* fix(web): timezone handling in search filter by @michelheusschen in https://github.com/immich-app/immich/pull/7384
* fix(web): prevent scroll reset on search page by @michelheusschen in https://github.com/immich-app/immich/pull/7385
* feat(web): Added password field visibility toggle by @Jan108 in https://github.com/immich-app/immich/pull/7368
* refactor(web): search box by @danieldietzler in https://github.com/immich-app/immich/pull/7397
* fix(web): count hidden people by @martabal in https://github.com/immich-app/immich/pull/7417
* fix(web): fetch error reporting by @michelheusschen in https://github.com/immich-app/immich/pull/7391
* fix(web): alignment of people in search box by @michelheusschen in https://github.com/immich-app/immich/pull/7430
* fix(web): prevent resetting date input when entering 0 by @michelheusschen in https://github.com/immich-app/immich/pull/7415
* fix(web): shared links page broken by enhanced:img by @michelheusschen in https://github.com/immich-app/immich/pull/7453
* fix(web): don't ask password for invalid shared link by @michelheusschen in https://github.com/immich-app/immich/pull/7456
* chore(web): cleanup promise handling by @benmccann in https://github.com/immich-app/immich/pull/7382
* chore(web): remove album's action notification by @alextran1502 in https://github.com/immich-app/immich/pull/7467
* chore(web): upgrade dependencies by @benmccann in https://github.com/immich-app/immich/pull/7471
### 🧠 Machine Learning
* Fix Smart Search when using OpenVINO by @agrawalsourav98 in https://github.com/immich-app/immich/pull/7389
* fix(ml): only use openvino if a gpu is available by @mertalev in https://github.com/immich-app/immich/pull/7450
### 📓 Documentation
* docs: add star history for README by @wangrongding in https://github.com/immich-app/immich/pull/7328
* Update reverse-proxy.md - Apache by @sigulete in https://github.com/immich-app/immich/pull/7386
### 🔨 Maintenance
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/7225
* chore(deps): update dependency vite to v5.1.3 by @renovate in https://github.com/immich-app/immich/pull/7247
* chore(deps): update dependency @types/node to v20.11.19 by @renovate in https://github.com/immich-app/immich/pull/7239
* chore(deps): update dependency @types/node to v20.11.19 by @renovate in https://github.com/immich-app/immich/pull/7238
* chore(deps): update dependency @types/node to v20.11.19 by @renovate in https://github.com/immich-app/immich/pull/7236
* chore(deps): update @immich/cli by @renovate in https://github.com/immich-app/immich/pull/7235
* fix(deps): update exiftool by @renovate in https://github.com/immich-app/immich/pull/7230
* refactor(server): e2e by @jrasm91 in https://github.com/immich-app/immich/pull/7265
* refactor: album e2e by @jrasm91 in https://github.com/immich-app/immich/pull/7320
* refactor: e2e client by @jrasm91 in https://github.com/immich-app/immich/pull/7324
* chore(deps): update base-image to v20240222 (major) by @renovate in https://github.com/immich-app/immich/pull/7338
* chore(deps): update vitest monorepo to v1.3.1 by @renovate in https://github.com/immich-app/immich/pull/7407
* chore(deps): update machine-learning by @renovate in https://github.com/immich-app/immich/pull/7451
* chore(ci): move e2e test to github runner by @alextran1502 in https://github.com/immich-app/immich/pull/7452
* chore(deps): update dependency @oazapfts/runtime to v1.0.1 by @renovate in https://github.com/immich-app/immich/pull/7446
* chore(deps): bump docker/setup-buildx-action from 3.0.0 to 3.1.0 by @dependabot in https://github.com/immich-app/immich/pull/7459
* chore(deps): update node.js to f3299f1 by @renovate in https://github.com/immich-app/immich/pull/7443
* chore(deps): update node.js to f3299f1 by @renovate in https://github.com/immich-app/immich/pull/7444
* fix(deps): update dependency orjson to v3.9.15 [security] by @renovate in https://github.com/immich-app/immich/pull/7438
* chore(deps): update server by @renovate in https://github.com/immich-app/immich/pull/7447
* fix(web/server): revert renovate hash update by @alextran1502 in https://github.com/immich-app/immich/pull/7464
* chore(deps): update @immich/cli by @renovate in https://github.com/immich-app/immich/pull/7445
* chore(deps): update web by @renovate in https://github.com/immich-app/immich/pull/7448
* chore(deps): update base-image to v20240227 (major) by @renovate in https://github.com/immich-app/immich/pull/7470
* chore(deps): update dependency @types/node to v20.11.20 by @renovate in https://github.com/immich-app/immich/pull/7473
* refactor(server): e2e by @jrasm91 in https://github.com/immich-app/immich/pull/7462
* chore(deps): update dependency @types/node to v20.11.20 by @renovate in https://github.com/immich-app/immich/pull/7475
* chore(deps): update dependency @types/node to v20.11.20 by @renovate in https://github.com/immich-app/immich/pull/7476

## New Contributors
* @MarcelEeken made their first contribution in https://github.com/immich-app/immich/pull/7305
* @wangrongding made their first contribution in https://github.com/immich-app/immich/pull/7328
* @RavidYael made their first contribution in https://github.com/immich-app/immich/pull/7339
* @sebbi08 made their first contribution in https://github.com/immich-app/immich/pull/7372
* @hrdl-github made their first contribution in https://github.com/immich-app/immich/pull/6907
* @AndyPro720 made their first contribution in https://github.com/immich-app/immich/pull/7359
* @Jan108 made their first contribution in https://github.com/immich-app/immich/pull/7368
* @sigulete made their first contribution in https://github.com/immich-app/immich/pull/7386
* @agrawalsourav98 made their first contribution in https://github.com/immich-app/immich/pull/7389

**Full Changelog**: https://github.com/immich-app/immich/compare/v1.95.1...

<hr /><em>This discussion was created from the release <a href='https://github.com/immich-app/immich/releases/tag/v1.96.0'>v1.96.0</a>.</em>

**Comment by Isrofilov** [comment]
```
docker exec -u postgres immich_postgres psql -d immich -c "CREATE EXTENSION IF NOT EXISTS pg_trgm;"
docker exec -u postgres immich_postgres psql -d immich -c "CREATE EXTENSION IF NOT EXISTS unaccent;"
```

**Reply by afermar** [reply]
After running this commands i get this: 
```
NOTICE:  extension "pg_trgm" already exists, skipping
NOTICE:  extension "unaccent" already exists, skipping
```
And the server still does not start properly.

**Reply by Isrofilov** [reply]
What database image are you using?
Check the current one [here](https://github.com/immich-app/immich/blob/2e56e777ce27f6a8b8dce6f37d2e5a872fb0fbf1/docker/docker-compose.yml#L68)

**Comment by dsm1212** [comment]
Ok, I'm probably just not finding this. My slideshow is always playing backwards in time. Can I reverse it? Actually I'd probably rather have the timeline reversed because I don't really like going backwards through the photos manually either.

steve

**Reply by alextran1502** [reply]
There is an option to randomize. Otherwise, it will go backward in time. Having an option to go forward in time is a good idea

**Reply by dsm1212** [reply]
I added a separate discussion/feature request. thx
https://github.com/immich-app/immich/discussions/7489

**Comment by mmomjian** [comment]
@alextran1502 greatly appreciate making it possible to run Immich in a shared DB without SU perms. In this case, the manual intervention was not needed. On both extension manual pages, the following appears. I was able to upgrade directly to 1.96.0 without adding the extensions and they now appear in the Immich database as added. 

> This module is considered “trusted”, that is, it can be installed by non-superusers who have CREATE privilege on the current database.

It’s not possible to run Immich without CREATE perms so in this case the upgrade should work fine for all users. 

[source](https://www.postgresql.org/docs/current/unaccent.html)

**Reply by mertalev** [reply]
Oh wow, thanks for mentioning this! I didn't know you could install trusted extensions without SU permission.

**Reply by mmomjian** [reply]
Sure thing! Hopefully this makes this and future database changes a little more frictionless.

**Comment by stephen304** [comment]
Is there anything special needed or a new android app update that hasn't fully rolled out for the blurred loading? I just updated to 1.96.0 and the android app for me still goes straight from grey to fully loaded - it also goes back to grey if I scroll away and back even if it's just a split second - I would think scrolling over the same section would cache something, be it the blurred thumb or the full image staying loaded:

https://github.com/immich-app/immich/assets/981915/578cf43f-54c3-43f1-9fe7-3a777b486e0c

**Comment by eil047** [comment]
I see and appreciate a lot of effort in the search possibilities, but unless I'm missing something, still no search possible on the item filename? I would assume this as the most basic/simple search operation? And it was available in the past? Why is it still missing? Thanks for your feedback!
